File: examples/QtDesigner/cpp/mainwindow.cpp
#include "mainwindow.h"
#include <Inventor/nodes/SoSeparator.h>
#include <Inventor/nodes/SoRotationXYZ.h>
#include <Inventor/nodes/SoMaterial.h>
#include <Inventor/nodes/SoCone.h>
#include <Inventor/engines/SoGate.h>
#include <Inventor/engines/SoElapsedTime.h>
#include <Inventor/fields/SoMFFloat.h>
#include <Inventor/Qt/viewers/SoQtExaminerViewer.h>

MainWindow::MainWindow(QWidget *parent)
  :QMainWindow(parent)
{
  setupUi(this);
  setupSoQt();
}

void
MainWindow::setupSoQt()
{
  SoSeparator *root = new SoSeparator();
  SoRotationXYZ *rotxyz = new SoRotationXYZ();
  SoGate *gate = new SoGate(SoMFFloat::getClassTypeId());
  SoElapsedTime *elapsedTime = new SoElapsedTime();
  gate->enable = false;
  gate->input->connectFrom(&elapsedTime->timeOut);
  rotxyz->angle.connectFrom(gate->output);
  SoMaterial *material = new SoMaterial();
  material->diffuseColor = SbColor(0.0, 1.0, 1.0);
  SoCone *cone = new SoCone();
  root->addChild(rotxyz);
  root->addChild(material);
  root->addChild(cone);
    
  this->exam = new SoQtExaminerViewer(this->examiner);
  this->exam->setSceneGraph(root);
}

File: examples/QtDesigner/python/mainwindow.py
#!/usr/bin/env python

from pivy.coin import *
from pivy.gui.soqt import SoQtExaminerViewer
from ui_test import Ui_MainWindow
from PyQt4.QtGui import QWidget
from PyQt4.QtGui import QMainWindow

class MainWindow(QMainWindow, Ui_MainWindow):
    def __init__(self, parent = None):
        QMainWindow.__init__(self, parent)
        self.setupUi(self)
        self.setupSoQt()

    def setupSoQt(self):
        root = SoSeparator()
        self.rotxyz = SoRotationXYZ()
        self.gate = SoGate(SoMFFloat.getClassTypeId())
        self.elapsedTime = SoElapsedTime()
        self.gate.enable = False
        self.gate.input.connectFrom(self.elapsedTime.timeOut)
        self.rotxyz.angle.connectFrom(self.gate.output)        
        self.material = SoMaterial()
        self.material.diffuseColor = (0.0, 1.0, 1.0)
        self.cone = SoCone()
        root.addChild(self.rotxyz)
        root.addChild(self.material)
        root.addChild(self.cone)

        self.exam = SoQtExaminerViewer(self.examiner)
        self.exam.setSceneGraph(root)
        #self.exam.show()
